src/Reflex/Process.hs

```diff
@@ -96,30 +96,27 @@ createRedirectedProcess mkWriteStdInput mkReadStdOutput mkReadStdError p (Proces
           Just pid -> do
             P.signalProcess sig pid >> return (Just sig)
       let
-          output :: Handle -> m (Event t o, ThreadId)
-          output h = do
-            (e, trigger) <- newTriggerEvent
-            reader <- liftIO $ mkReadStdOutput h trigger
-            t <- liftIO $ forkIO reader
-            return (e, t)
+        output :: Handle -> m (Event t o, Async ())
+        output h = do
+          (e, trigger) <- newTriggerEvent
+          reader <- liftIO $ mkReadStdOutput h trigger
+          t <- liftIO $ async reader
+          return (e, t)
+
+        err_output :: Handle -> m (Event t e, Async ())
+        err_output h = do
+          (e, trigger) <- newTriggerEvent
+          reader <- liftIO $ mkReadStdError h trigger
+          t <- liftIO $ async reader
+          return (e, t)
 
-      let
-          err_output :: Handle -> m (Event t e, ThreadId)
-          err_output h = do
-            (e, trigger) <- newTriggerEvent
-            reader <- liftIO $ mkReadStdError h trigger
-            t <- liftIO $ forkIO reader
-            return (e, t)
       (out, outThread) <- output hOut
       (err, errThread) <- err_output hErr
       (ecOut, ecTrigger) <- newTriggerEvent
-      void $ liftIO $ forkIO $ do
+      void $ liftIO $ async $ flip finally (P.cleanupProcess po) $ do
         waited <- waitForProcess ph
-        mask_ $ do
-          ecTrigger waited
-          P.cleanupProcess po
-          killThread outThread
-          killThread errThread
+        _ <- waitBoth outThread errThread
+        ecTrigger waited -- Output events should never fire after process completion
```

Comment:
This may be more convenient in many cases but it's also slightly less "honest".

Reply:
However without this we'd probably need a *second* event that fires when there is no output left to process.

Reply:
We should use 2 events instead.

Reply:
Need to document this invariant.